numba/core/types/containers.py

```diff
@@ -642,6 +673,45 @@ def unify(self, typingctx, other):
             if not other.is_precise():
                 return self
 
+    @property
+    def key(self):
+        return self.key_type, self.value_type, str(self.initial_value)
+
+
+class LiteralStrKeyDict(Literal, NamedTuple):
+    def __init__(self, literal_value, value_index=None):
+        self._literal_init(literal_value)
+        self.value_index = value_index
+        from collections import namedtuple
+        strkeys = [x.literal_value for x in literal_value.keys()]
+        self.tuple_ty = namedtuple('_ntclazz', ' '.join(strkeys))
+        self.tuple_inst = self.tuple_ty(*literal_value.values())
+        tys = [unliteral(x) for x in literal_value.values()]
+        NamedTuple.__init__(self, tys, self.tuple_ty)
+        self.name = 'LiteralStrKey[Dict]({})'.format(literal_value)
+        literal_vals = [getattr(x, 'literal_value', None) for x in literal_value.values()]
+
+    def __unliteral__(self):
+        return self
```

Comment:
Couldn't this degrade into a dict with initial_value?

Reply:
May be, same issue as https://github.com/numba/numba/pull/5946/files#r450826105

Reply:
Same as https://github.com/numba/numba/pull/5946/files/3e32c6e662f01ad9cae73c36ead947dd82c1000b#r451469788